### PR TITLE
Slack webhook channel name cannot be empty or just contain an hashtag

### PR DESCRIPTION
**Pull request by adelowo**
- [x] Don't allow an empty string to be used as a slack channel name.
- [x] Don't allow just # to be used as a slack channel name.. Slack doesn't allow this itself in the app
- [x] Don't allow channel name to contain spaces e.g ` . channel  `

> There can also be another added check which I have skipped... channel names cannot contain spaces in between or punctuations e.g `ch nae` or `cd##jndd`..  Should that be added too or this is good enough ?

### CODE REVIEW

File: routers/api/v1/utils/hook.go [hunk 1]
```diff
@@ -5,14 +5,16 @@
...
 	"encoding/json"
 	"net/http"
+	"strings"
+
+	api "code.gitea.io/sdk/gitea"
 
 	"code.gitea.io/gitea/models"
 	"code.gitea.io/gitea/modules/context"
 	"code.gitea.io/gitea/routers/api/v1/convert"
+	"code.gitea.io/gitea/routers/utils"
```
Comment (appleboy) [inline routers/api/v1/utils/hook.go:15]: add an empty line between internal and external package.

**Review by appleboy — CHANGES_REQUESTED** [review]

File: routers/api/v1/utils/hook.go (hunk 1 above)
Comment (adelowo) [reply routers/api/v1/utils/hook.go:15]: Done @appleboy 968d1f2274dc11ab73f0e670d68a0a56a950f238 .. Not sure why this didn't get collapsed

**Review by lafriks — APPROVED** [review]

**Comment by techknowlogick** [comment]
@appleboy pending your review

File: routers/api/v1/utils/hook.go [hunk 2]
```diff
@@ -5,14 +5,17 @@
 package utils
 
 import (
-	api "code.gitea.io/sdk/gitea"
-
 	"encoding/json"
 	"net/http"
+	"strings"
+
+	api "code.gitea.io/sdk/gitea"
```
Comment (appleboy) [inline routers/api/v1/utils/hook.go]: remove empty line between internal package.

**Review by appleboy — CHANGES_REQUESTED** [review]

File: routers/api/v1/utils/hook.go (hunk 2 above)
Comment (appleboy) [reply routers/api/v1/utils/hook.go]: otherwise L-GTM

Comment (appleboy) [reply routers/api/v1/utils/hook.go]: remove line13

**Comment by lafriks** [comment]
@appleboy need your approval

**Review by appleboy — APPROVED** [review]

**Pull request merged**